Hi @inetknght , thank you very much for the pull request. I'm trying to build your branch locally with g++ version 7.5.0 , but apart from that deprecated header warning I'm also getting compilation errors:

```
In file included from /home/inetic/boost/boost_1_74_0/boost/coroutine/exceptions.hpp:14:0,
                 from /home/inetic/boost/boost_1_74_0/boost/coroutine/asymmetric_coroutine.hpp:24,
                 from /home/inetic/boost/boost_1_74_0/boost/coroutine/coroutine.hpp:10,
                 from /home/inetic/boost/boost_1_74_0/boost/coroutine/all.hpp:11,
                 from /home/inetic/boost/boost_1_74_0/boost/asio/spawn.hpp:19,
                 from /home/inetic/cpp-upnp/example/../include/upnp/ssdp.h:11,
                 from /home/inetic/cpp-upnp/src/igd.cpp:8:
/home/inetic/boost/boost_1_74_0/boost/detail/scoped_enum_emulation.hpp:17:55: note: #pragma message: This header is deprecated. Use <boost/core/scoped_enum.hpp> instead.
 BOOST_HEADER_DEPRECATED("<boost/core/scoped_enum.hpp>")
                                                       ^
In file included from /usr/include/c++/7/memory:64:0,
                 from /home/inetic/boost/boost_1_74_0/boost/asio/detail/memory.hpp:19,
                 from /home/inetic/boost/boost_1_74_0/boost/asio/execution/detail/as_invocable.hpp:20,
                 from /home/inetic/boost/boost_1_74_0/boost/asio/execution/execute.hpp:20,
                 from /home/inetic/boost/boost_1_74_0/boost/asio/execution/executor.hpp:20,
                 from /home/inetic/boost/boost_1_74_0/boost/asio/execution/allocator.hpp:20,
                 from /home/inetic/boost/boost_1_74_0/boost/asio/execution.hpp:18,
                 from /home/inetic/boost/boost_1_74_0/boost/asio/any_io_executor.hpp:22,
                 from /home/inetic/boost/boost_1_74_0/boost/asio/basic_socket_acceptor.hpp:19,
                 from /home/inetic/boost/boost_1_74_0/boost/asio/ip/tcp.hpp:19,
                 from /home/inetic/cpp-upnp/src/igd.cpp:1:
/usr/include/c++/7/bits/stl_construct.h: In instantiation of ‘void std::_Construct(_T1*, _Args&& ...) [with _T1 = upnp::igd; _Args = {const upnp::igd&}]’:
/usr/include/c++/7/bits/stl_uninitialized.h:83:18:   required from ‘static _ForwardIterator std::__uninitialized_copy<_TrivialValueTypes>::__uninit_copy(_InputIterator, _InputIterator, _ForwardIterator) [with _InputIterator = __gnu_cxx::__normal_iterator<const upnp::igd*, std::vector<upnp::igd> >; _ForwardIterator = upnp::igd*; bool _TrivialValueTypes = false]’
/usr/include/c++/7/bits/stl_uninitialized.h:134:15:   required from ‘_ForwardIterator std::uninitialized_copy(_InputIterator, _InputIterator, _ForwardIterator) [with _InputIterator = __gnu_cxx::__normal_iterator<const upnp::igd*, std::vector<upnp::igd> >; _ForwardIterator = upnp::igd*]’
/usr/include/c++/7/bits/stl_uninitialized.h:289:37:   required from ‘_ForwardIterator std::__uninitialized_copy_a(_InputIterator, _InputIterator, _ForwardIterator, std::allocator<_Tp>&) [with _InputIterator = __gnu_cxx::__normal_iterator<const upnp::igd*, std::vector<upnp::igd> >; _ForwardIterator = upnp::igd*; _Tp = upnp::igd]’
/usr/include/c++/7/bits/stl_vector.h:331:31:   required from ‘std::vector<_Tp, _Alloc>::vector(const std::vector<_Tp, _Alloc>&) [with _Tp = upnp::igd; _Alloc = std::allocator<upnp::igd>]’
/home/inetic/boost/boost_1_74_0/boost/outcome/detail/value_storage.hpp:857:37:   required from ‘boost::outcome_v2::detail::value_storage_nontrivial<T>::value_storage_nontrivial(boost::outcome_v2::in_place_type_t<T>, Args&& ...) [with Args = {std::vector<upnp::igd, std::allocator<upnp::igd> >&}; T = std::vector<upnp::igd>]’
/home/inetic/boost/boost_1_74_0/boost/outcome/detail/value_storage.hpp:990:17:   required from ‘constexpr boost::outcome_v2::detail::basic_result_storage<R, EC, NoValuePolicy>::basic_result_storage(boost::outcome_v2::in_place_type_t<typename std::conditional<std::is_same<_From, _To>::value, boost::outcome_v2::detail::basic_result_storage<R, EC, NoValuePolicy>::disable_in_place_value_type, R>::type>, Args&& ...) [with Args = {std::vector<upnp::igd, std::allocator<upnp::igd> >&}; R = std::vector<upnp::igd>; EC = boost::system::error_code; NoValuePolicy = boost::outcome_v2::policy::terminate; typename std::conditional<std::is_same<_From, _To>::value, boost::outcome_v2::detail::basic_result_storage<R, EC, NoValuePolicy>::disable_in_place_value_type, R>::type = std::vector<upnp::igd>]’
/home/inetic/boost/boost_1_74_0/boost/outcome/detail/basic_result_value_observers.hpp:44:17:   required from ‘constexpr boost::outcome_v2::basic_result<R, S, NoValuePolicy>::basic_result(T&&, boost::outcome_v2::basic_result<R, S, NoValuePolicy>::value_converting_constructor_tag) [with T = std::vector<upnp::igd>&; <template-parameter-2-2> = void; R = std::vector<upnp::igd>; S = boost::system::error_code; NoValuePolicy = boost::outcome_v2::policy::terminate]’
/home/inetic/cpp-upnp/src/igd.cpp:373:12:   required from here
/usr/include/c++/7/bits/stl_construct.h:75:7: error: use of deleted function ‘upnp::igd::igd(const upnp::igd&)’
     { ::new(static_cast<void*>(__p)) _T1(std::forward<_Args>(__args)...); }
       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/inetic/cpp-upnp/src/igd.cpp:11:0:
/home/inetic/cpp-upnp/example/../include/upnp/igd.h:22:7: note: ‘upnp::igd::igd(const upnp::igd&)’ is implicitly declared as deleted because ‘upnp::igd’ declares a move constructor or move assignment operator
 class igd final {
       ^~~
CMakeFiles/cpp_upnp.dir/build.make:75: recipe for target 'CMakeFiles/cpp_upnp.dir/home/inetic/cpp-upnp/src/igd.cpp.o' failed
make[2]: *** [CMakeFiles/cpp_upnp.dir/home/inetic/cpp-upnp/src/igd.cpp.o] Error 1
CMakeFiles/Makefile2:104: recipe for target 'CMakeFiles/cpp_upnp.dir/all' failed
make[1]: *** [CMakeFiles/cpp_upnp.dir/all] Error 2
Makefile:83: recipe for target 'all' failed
make: *** [all] Error 2
```

I wonder why CCI isn't triggering for PRs...

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -17,8 +17,8 @@ jobs:
           command: |
             apt-get update
             apt-get install -y \
-                libboost1.71-dev \
-                libboost-coroutine1.71-dev \
+                libboost1.74-dev \
+                libboost-coroutine1.74-dev \
                 build-essential \
                 cmake \
                 ninja-build \

--- a/src/igd.cpp
+++ b/src/igd.cpp
@@ -91,7 +91,7 @@ igd::get_external_address(net::yield_context yield) noexcept
     auto addr = net::ip::make_address(ip_s, ec);
     if (ec) return error::bad_address{};
 
-    return addr;
+    return {std::move(addr)};
 }
 
 result<igd::map_entry, igd::error::get_generic_port_mapping_entry>
@@ -212,7 +212,7 @@ igd::get_list_of_port_mappings( protocol proto
                           , bool(*oena)});
     }
 
-    return entries;
+    return {std::move(entries)};
 }
 
 result<void, igd::error::delete_port_mapping>
@@ -290,7 +290,7 @@ igd::soap_request( string_view command
         return E{error::http_status{rs.result()}};
     }
 
-    return rs;
+    return {std::move(rs)};
 }
 
 /* static */
@@ -370,7 +370,7 @@ result<std::vector<igd>> igd::discover(net::any_io_executor exec, net::yield_con
         }
     }
 
-    return igds;
+    return {std::move(igds)};
 }
 
 /* static */
@@ -418,7 +418,7 @@ igd::query_root_device( net::any_io_executor exec
     auto opt_root_dev = device_parse_root(rs.body());
     if (!opt_root_dev) return sys::errc::io_error;
 
-    return std::move(*opt_root_dev);
+    return {std::move(*opt_root_dev)};
 }
 
 void igd::stop() {

--- a/src/ssdp.cpp
+++ b/src/ssdp.cpp
@@ -149,7 +149,7 @@ query::state_t::get_response(net::yield_context yield)
 
     auto r = std::move(_responses.front());
     _responses.pop();
-    return std::move(r.value());
+    return {std::move(r.value())};
 }
 
 /* static */
@@ -216,7 +216,7 @@ query::response::parse(string_view lines)
         }
     }
 
-    return ret;
+    return {std::move(ret)};
 }
 
 void query::state_t::stop() {

